### ISSUE TITLE
 Error in orderCells: 'from' contains NAs

### ISSUE THREAD

**Issue by rblame** [cole-trapnell-lab/monocle-release#518]
I encountered an error while trying to use the orderCells function from the monocle package. The error message is as follows:

```
Error in sequence.default(n..1, from = seq.int(s.1, length(df), s.1), :
'from' contains NAs
```

Steps to Reproduce:

Load the monocle package:

`library(monocle)`

Load the RData file containing the CellDataSet object:

`load("cds1.RData")`

Attempt to run orderCells on the CellDataSet object:

`cds <- orderCells(cds)`

Expected Behavior:

The function should successfully order the cells based on the trajectory analysis without errors.

Actual Behavior:

The function throws an error indicating that 'from' contains NAs. This suggests that there may be missing values in the data that the function is unable to handle properly.

Data Example:

Unfortunately, I cannot provide the exact data due to its sensitivity and size, but it is a standard CellDataSet object used for single-cell trajectory analysis.

Environment:

R version: 4.4
monocle package version: 2.24
Operating System: Linux

**Comment by whylassan** [comment]
I got the same issue. 
No clues at all.

**Comment by Alexxyz123** [comment]
I also found this problem, and I noticed that the warnings when I conduct  
mycds <- estimateDispersions(mycds, cores=16)
Removing 120 outliers

Warning messages:
1: `group_by_()` was deprecated in dplyr 0.7.0.
ℹ Please use `group_by()` instead.
ℹ See vignette('programming') for more help
ℹ The deprecated feature was likely used in the monocle package.
  Please report the issue to the authors.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this
warning was generated. 
2: `select_()` was deprecated in dplyr 0.7.0.
ℹ Please use `select()` instead.
ℹ The deprecated feature was likely used in the monocle package.
  Please report the issue to the authors.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this
warning was generated. 
3: In log(ifelse(y == 0, 1, y/mu)) : NaNs produced
4: step size truncated due to divergence 
I wondering if this step produced NA for the ordercells?

**Comment by huipan1973** [comment]
Same error. Any suggestion?

**Comment by Alexxyz123** [comment]
No, I can't find the solution

**Comment by lzh-fugwaa** [comment]
I have the same problem. Have you solved it?

**Comment by huipan1973** [comment]
No, I haven't. I tried to filter genes and cells with more stringent cutoff. But I still got lots of NAs in the pseudotime estimation.

**Comment by Alexxyz123** [comment]
Yes， I also filter genes and cells with more stringent cutoff. But also get the error. However another dataset did not meet this error. How can I check the difference between the two datasets. Maybe the names of samples afre not right? it should be "_" not "-"

**Comment by Woohootakeoff** [comment]
I had the same problem. I found that the error occurred on line 42 of the orderCell function "dp < -as.matrix (dist(t(P)))". There was an error when calculating dist. I am not sure if it is my data or the R package version. My version of R is 4.4.1, monocle 2.32.0, and stats 4.4.1.
<img width="1230" alt="P" src="https://github.com/user-attachments/assets/50b8f16b-baef-4d51-a201-0a1e18cfb5d6">

**Comment by lzh-fugwaa** [comment]
Thank you for your reply! I tried again later.Is it because there is too much data?I filter out some barcode about 60000 counts,but also Report the same error



---- Replied Message ----
| From | ***@***.***> |
| Date | 09/30/2024 17:30 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [cole-trapnell-lab/monocle-release] Error in orderCells: 'from' contains NAs (Issue #518) |

I had the same problem. I found that the error occurred on line 42 of the orderCell function "dp < -as.matrix (dist(t(P)))". There was an error when calculating dist. I am not sure if it is my data or the R package version. My version of R is 4.4.1, monocle 2.32.0, and stats 4.4.1.
P.png (view on web)

**Comment by rblame** [comment]
Has any friend solved this problem?

**Comment by Alexxyz123** [comment]
Anybody solved this problem?

**Comment by Ares1314** [comment]
I encountered the same issue and tried downgrading the version and reinstalling, but neither worked. Does anyone know how to resolve this? Please help, thank you!

**Comment by wujian-njau** [comment]
me too, When I randomly sampled 3000 cells, it worked fine

**Comment by earthislarge** [comment]
Same here.. Couldn't know what;s going on

**Comment by cathy-321** [comment]
Same as meeeee！ Any solution，i even cant find the object “s.1” in the original code!

**Comment by dhaefh** [comment]
yes.When I randomly sampled 3000 cells, it worked fine

**Comment by SvenjaRoo** [comment]
I made a similar observation. When I execute semivariogram() from the geostats package with a large data set (65,536 rows), I get:

```
Error in sequence.default(n..1, from = seq.int(s.1, length(df), s.1), : 
'from' contains NAs
Additionally: Warning message:
In sequence.default(n..1, from = seq.int(s.1, length(df), s.1), :
NAs generated by coercion to integer range
```

It runs without error messages when I restrict the data set. Somewhere between 46,225 (=215²) and 46,656 (=216²) rows is the point where it breaks.
Maybe the additional warning message helps someone to trace back the problem?

R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows Server 2019 x64 (build 17763)
geostats_1.6

**Comment by louhao1129** [comment]
I encountered the same NAs introduced by coercion to integer range error when running orderCells() on a large dataset（>60000 cells).
```
In sequence.default(n..1, from = seq.int(s.1, length(df), s.1),  :
  NAs introduced by coercion to integer range
```

 After investigation, I found this is caused by R integer overflow when the internal matrix dimensions exceed .Machine$integer.max (~2.1 billion).

Solutions (verified working) : Subsample cells (quickest fix)
```
set.seed(42)
cells_to_keep <- sample(colnames(seurat_obj), size = 30000)
seurat_sub <- subset(seurat_obj, cells = cells_to_keep)
```

Option 2: Switch to Monocle3 (recommended for large datasets)

**Comment by LvEF-2001** [comment]
I ran `orderCells(cds)` with 60,000 cells and encountered the same error. Then I consulted GPT, which suggested that the large number of cells might have prevented some cells from being stably mapped to graph nodes during the DDRTree process, potentially causing `NA` values in the subsequent `orderCells()` function. 
I successfully ran the program after reducing the number of cells to under 40,000 through stratified sampling.
